pathd/path_pcep_lib.c

```diff
@@ -0,0 +1,1205 @@
+	config->support_pce_lsp_instantiation = false;
+	config->support_include_db_version = false;
+	config->support_lsp_triggered_resync = false;
+	config->support_lsp_delta_sync = false;
+	config->support_pce_triggered_initial_sync = false;
+	config->support_sr_te_pst = true;
+	config->pcc_can_resolve_nai_to_sid = false;
+
+	config->max_sid_depth = msd;
+	config->pcep_msg_versioning->draft_ietf_pce_segment_routing_07 =
+		pcep_options->draft07;
+	config->keep_alive_seconds = pcep_options->keep_alive_seconds;
+	config->min_keep_alive_seconds = pcep_options->min_keep_alive_seconds;
+	config->max_keep_alive_seconds = pcep_options->max_keep_alive_seconds;
+	config->dead_timer_seconds = pcep_options->dead_timer_seconds;
+	config->min_dead_timer_seconds = pcep_options->min_dead_timer_seconds;
+	config->max_dead_timer_seconds = pcep_options->max_dead_timer_seconds;
+	config->request_time_seconds = pcep_options->pcep_request_time_seconds;
+	/* TODO when available in the pceplib, set it here
+	 pcep_options->state_timeout_inteval_seconds;*/
+
+	if (pcep_options->tcp_md5_auth != NULL
+	    && pcep_options->tcp_md5_auth[0] != '\0') {
+		config->is_tcp_auth_md5 = true;
+		strncpy(config->tcp_authentication_str,
+			pcep_options->tcp_md5_auth, TCP_MD5SIG_MAXKEYLEN);
+	} else {
+		config->is_tcp_auth_md5 = false;
+	}
+
+	if (IS_IPADDR_V6(dst_addr)) {
+		sess = connect_pce_ipv6(config, &dst_addr->ipaddr_v6);
+	} else {
+		sess = connect_pce(config, &dst_addr->ipaddr_v4);
+	}
+	destroy_pcep_configuration(config);
+	return sess;
+}
+
+void pcep_lib_disconnect(pcep_session *sess)
+{
+	assert(sess != NULL);
+	disconnect_pce(sess);
+}
+
+/* Callback passed to pceplib to create a timer.
+ * When the timer expires, pcep_lib_timer_expire() will be called */
+
+/* Commenting out until issue with FRR timer cancellation resolved:
+ * https://gitlab.voltanet.io/volta-controlplane/pceplib/-/issues/152
+void pcep_lib_pceplib_timer_create_cb(void *fpt, void **thread, int delay,
+				      void *payload)
+{
+	struct ctrl_state *ctrl_state = ((struct frr_pthread *) fpt)->data;
+
+	pcep_thread_schedule_pceplib_timer(
+		ctrl_state, delay, payload, (struct thread **) thread,
+		pcep_lib_timer_expire);
+}
+*/
+
+/* Callback passed to pceplib to cancel a timer */
+
+/* Commenting out until issue with FRR timer cancellation resolved:
+ * https://gitlab.voltanet.io/volta-controlplane/pceplib/-/issues/152
+void pcep_lib_pceplib_timer_cancel_cb(void **thread)
+{
+	pcep_thread_cancel_timer((struct thread **)thread);
+}
+*/
+
+/* Callback called by path_pcep_controller when a timer expires */
+
+/* Commenting out until issue with FRR timer cancellation resolved:
+ * https://gitlab.voltanet.io/volta-controlplane/pceplib/-/issues/152
+int pcep_lib_timer_expire(struct thread *thread)
+{
+    struct pcep_ctrl_timer_data *data = THREAD_ARG(thread);
+    assert(data != NULL);
+
+    pceplib_external_timer_expire_handler(data->payload);
+
+    XFREE(MTYPE_PCEP, data);
+
+    return 0;
+}
+*/
```

Comment:
Can you remove these commented functions ? And re-introduce them latter with a dedicated PR ?